Modules/_cursesmodule.c

```diff
@@ -3271,9 +3274,14 @@ _curses_initscr_impl(PyObject *module)
 #define SetDictInt(string,ch)                                           \
     do {                                                                \
         PyObject *o = PyLong_FromLong((long) (ch));                     \
-        if (o && PyDict_SetItemString(ModDict, string, o) == 0)     {   \
+        if (o == NULL) {                                                \
+            goto error;                                                 \
+        }                                                               \
+        if (PyDict_SetItemString(d, string, o) < 0) {                   \
             Py_DECREF(o);                                               \
+            goto error;                                                 \
```

Comment:
"return NULL;" is enough: you should remove the error label. An error label is used when you need to clean up things, it's not the case here. Same remark at the bottom of the file.